src/localization/locale/en.ts

```diff
@@ -101,6 +101,7 @@ export const en = {
     artifacts: 'Artifacts',
     artifactSearch: 'Search artifacts',
     variablesSearch: 'Search variables',
+    variableMustBeSnakeCase: 'Variable names may only contain letters, numbers, and underscores and may not begin or end with an underscore',
```

Comment:
If this was a generic snake case message then the validation rule could pretend the name field.